### PR TITLE
Create the first tab "DESCRIPTION" of the "Create Project" view

### PR DESCRIPTION
**Pull request by denektenina**
# Description

Link to Jira issue: [https://sikt.atlassian.net/browse/NP-47287](https://sikt.atlassian.net/browse/NP-47287)

 ⚠**This is only visible when you have beta toggled to true in local storage** ⚠

You get to this view my going to Min Side --> "Prosjektregistreringer" in the left panel -> Choose a project -> Click the edit-button

This is a part of the effort to create a new create project-wizard [(NP-47281)](https://sikt.atlassian.net/browse/NP-47281), and there is a subtask for each of the 4 steps, plus subtasks for some new functionality to add.

In this task we have to create the first tab, which also means wiring up some of the skeleton, like the steps-panel.

**What should work after this PR:**
* In projects view, it should now be possible to click the edit button and get into the new edit project wizard on the first panel:

image

* It should be possible to see the current values in the form, and change them. 
* If you remove mandatory fields and click away from the field, an error message should display under the field
* You can click on the stepper to change views (the other views are empty)
* If there are any errors on the page you are leaving, there will be an error icon in the stepper
* I made some of the registration-form-functions for checking error generic so verify that validation still works as it should in the registration form as well.
* This new view should only be visible when beta is toggled to true in localStorage, so verity that it looks as before when this is not the case.

**Things that are not working and will be fixed in later PRs:**
* There are no other views / panels yet
* There is no save-button
* There is no cancel-button
* There is no-next- or previous button
* The stepper is not ready for mobile-view yet
* There is still nothing when you open the new-project view
* The touched-logic between panels is not implemented

# Checklist

Ensure that the changes are aligned with the expectations of PO/designer before marking the PR as ready for review.

Also ensure that the following criterias are met:

- [x] The changes are working as expected
- [x] The changes are tested OK for different screen sizes
- [x] The changes are tested OK for a11y
- [x] Interactive elements have data-testids
- [x] I have done a QA of my own changes

_Note: some of these criterias may not always be relevant, and can simply be marked as completed._

### CODE REVIEW

File: src/utils/formik-helpers.ts [hunk 1]
```diff
@@ -34,11 +36,11 @@ export interface TabErrors {
   [RegistrationTab.FilesAndLicenses]: string[];
 }
 
-const getErrorMessages = (
-  fieldNames: string[],
-  errors: FormikErrors<Registration>,
-  touched?: FormikTouched<Registration>
-) => {
+export interface ProjectTabErrors {
```
Comment (elkebab) [inline src/utils/formik-helpers.ts:39]:
```suggestion
interface ProjectTabErrors {
```

File: src/utils/formik-helpers.ts [hunk 2]
```diff
@@ -34,11 +36,11 @@ export interface TabErrors {
...
+  [RegistrationTab.Description]: string[];
+}
+
+const getErrorMessages = <T>(fieldNames: string[], errors: FormikErrors<T>, touched?: FormikTouched<T>) => {
```
Comment (elkebab) [inline src/utils/formik-helpers.ts:43]: Nice! Egentlig litt overrasket over at dette var mulig å gjenbruke med så lite tilpasninger :D

File: src/utils/formik-helpers.ts [hunk 3]
```diff
@@ -75,6 +76,18 @@ export const getTabErrors = (
   return tabErrors;
 };
 
+export const getProjectTabErrors = (errors: FormikErrors<CristinProject>, touched?: FormikTouched<CristinProject>) => {
```
Comment (elkebab) [inline src/utils/formik-helpers.ts:79]: Allerede ganske stor denne fila, så det kan være at argument for å flytte alt prosjekt-spesifikt til `project-form-helpers` elns. Kanskje samme med registration også, sånn at det til slutt er:
- commonFormHelpers.ts
- registrationFormHelpers.ts
- projectFormHelpers.ts

**Review by elkebab — APPROVED** [review]
En del kode å sette seg inn i, så skumma bare gjennom. Ser ut som løsningen er ganske lik den for registration wizard, og det er jo noe som virker i dag da. Men som sagt er det nok mye som kan forenkles, så fullt lov å tenke nytt om noe er unødig tungvint/komplisert.

Du kan vurdere hvilke kommentarer du vil ta hensyn til nå eller senere. Fint at det blir flere mindre PRs, så greit å bare utsette ting så lenge det er i beta.

Alt i alt ser det bra ut!

File: src/utils/formik-helpers.ts (hunk 2 above)
Comment (denektenina) [reply src/utils/formik-helpers.ts:43]: Enig!

File: src/utils/formik-helpers.ts (hunk 3 above)
Comment (denektenina) [reply src/utils/formik-helpers.ts:79]: God idé! Jeg tror ikke jeg bør gjøre alt det i denne PR'en, for det medfører at vi må teste en del også i registreringsskjemaet. Men jeg henter ut alle som er for project, og lager en ny folder-struktur som klargjør for det.

**Review by elkebab — APPROVED** [review]

**Pull request merged**